### PR TITLE
Signup feature

### PR DESCRIPTION
**Pull request by clalil**
# Add sign up feature
[PT-link](https://www.pivotaltracker.com/story/show/168723553)

## This PR contains
- Signup feature
- Signup feature tests
- Factory bot for creating a mock user
- Unit tests for signup confirmation
- Sad paths feature testing

## What I have learned
Creating factory bots
Debugging factory bots
How to navigate the different folders
Creating unit tests for validating models
How to use Pry with Cucumber

### CODE REVIEW

**Comment by faraznaeem** [comment]
Wrong story in the pt link

File: Gemfile [hunk 1]
```diff
@@ -22,10 +22,11 @@ group :development, :test do
   gem 'factory_bot_rails'
   gem 'cucumber-rails', require: false
   gem 'database_cleaner'
+    gem 'pry-rails'
```
Comment (faraznaeem) [inline Gemfile]: indentation

File: config/routes.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
   devise_for :users
   get 'welcome/index'
 
-  root 'welcome#index'
+  root controller: :welcome, action: :index
```
Comment (faraznaeem) [inline config/routes.rb:6]: nice

File: features/sign_up.feature [hunk 3]
```diff
@@ -0,0 +1,37 @@
+Feature: Signup to the webmail
+As a visitor
```
Comment (faraznaeem) [inline features/sign_up.feature]: use proper indentation in this file

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: features/sign_up.feature (hunk 3 above)
Comment (clalil) [reply features/sign_up.feature]: Changes done accordingly. Thank you for your valid input.

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**